apps/src/lib/util/audioApi.js

```diff
@@ -173,7 +200,8 @@ export const commands = {
       gender,
       locale: voices[language].locale,
       authenticityToken,
-      onFailure: message => outputWarning(message + '\n')
+      onFailure: message => outputWarning(message + '\n'),
+      onComplete: validOnComplete && onComplete
```

Comment:
i think this will cause unexpected behavior when `validOnComplete` is false

```javascript
const func = () => {}

let obj = {onComplete: true && func}
obj.onComplete() // executes func

obj = {onComplete: false && func}
obj.onComplete() // Uncaught TypeError: obj.onComplete is not a function
```

Reply:
I think that's okay because it's optional for `AzureTextToSpeech.createSoundPromise`. So if `validOnComplete` is `false`, then `opts.onComplete` will just be `undefined`, but that's already handled in the `onEnded` function defined in `createSoundResponse_`

If there's a better way to do this, please lmk!

Reply:
maybe i'm missing something, but i don't think that case is handled in `createSoundResponse_`

```javascript
// audioApi.js
const promise = azureTTS.createSoundPromise({
  // other options
  onComplete: validOnComplete && onComplete // evaluates to false
});

// AzureTextToSpeech.js
createSoundResponse_ = opts => {
    const onEnded = () => {
      // opts.onComplete === false, so opts.onComplete?.() throws
      opts.onComplete?.();
      this.onSoundComplete_();
    };
}
```

if i'm understanding correctly, i think we have 2 options. if i'm not understanding correctly, ignore the stuff below and correct me 😄 
1. go back to not using optional chaining in AzureTextToSpeech
```javascript
if (opts.onComplete) {
  opts.onComplete();
}
```
2. only pass the `onComplete` option to `createSoundPromise` if `validOnComplete` is true, otherwise `onComplete` evaluates to undefined/null

i prefer option 2 because i think it's confusing for a config parameter to be either a function or false (if it's not a function, it should be undefined/null). in audioApi.js, we could instead do something like

```javascript
const promise = azureTTS.createSoundPromise({
  // other options
  onComplete: validOnComplete ? onComplete : null
});
```

Reply:
ah, I'm sorry you're totally right, not sure what I was thinking yesterday but what I wrote above is just totally incorrect. I think I agree, option 2 is better.